### ISSUE TITLE
blacksheep can't show debug when error occured

### ISSUE THREAD

**Issue by sugizo** [RobertoPrevato/BlackSheep#75]
blacksheep can't show debug when error occured (either embed in python file or execute the uvicorn server in terminal)
first think related with uvicorn, but test with another framework the same command or python uvicorn.run() can show debug on terminal when error occured (e.g. starlette, fastapi)
**e.g.**
**server.py**
```
import uvicorn
from music import *
from music.settings import app
if __name__ == "__main__":
    uvicorn.run("server:app", host = "0.0.0.0", port = 8000, log_level = "debug", reload = True)
```

**test with shell after comment the if condition to execute uvicorn.run()**
```
source activate python3_test
cd ~/Downloads/blacksheep
uvicorn server:app --host 0.0.0.0 --port 8000 --log-level debug --reload
```

**result on terminal**
INFO:     127.0.0.1:52791 - "PUT /api/put/instrument/2 HTTP/1.1" 500 Internal Server Error

**expected result**
have an error traceback in detail (debug level)

thanks

**Comment by RobertoPrevato** [comment]
Hi,
It is possible to return error details to the client using the option `show_error_details` when instantiating the `Application` object.
Please see the documentation [here](https://www.neoteroi.dev/blacksheep/application/#handling-errors).

You have a good point about using `logging.debug` to log these events, I will fix this in the next version of the framework.
Thank You for your feedback and useful comments!

**Comment by sugizo** [comment]
`app = Application(show_error_details = True)`
only result for show error details in web browser env, n think it's not good for visitor, perhaps the error record in backend log file or just show in terminal
or in other hand, if an error occured perhaps can learn some of idea about ticketing (store all error in sqlite database as ticket id n detail for further investigation n only administrator can show the error detail (need password) ) like in [web2py](web2py.com) or [py4web](py4web.com)

**Comment by RobertoPrevato** [comment]
You are right, I will apply the changes soon and release it with the next version of the framework.

**Comment by RobertoPrevato** [comment]
Thanks for the issue report, @sugizo. Now this problem has been corrected in `v0.3.2` and is available on PyPi.
I didn't verify how Starlette exactly handles this, but I modified the code to log both unhandled exceptions and user defined exceptions of `HTTPException` type (useful to handle some scenarios in which the user wants to throw exception).
